libarchive/test/test_read_format_7zip.c

```diff
@@ -1302,6 +1302,63 @@ DEFINE_TEST(test_read_format_7zip_extract_second)
 	assertEqualInt(ARCHIVE_OK, archive_read_free(a));
 }
 
+#ifdef HAVE_LZMA_FILTER_RISCV
+static void
+test_riscv_filter(const char *refname)
+{
+	struct archive *a;
+	struct archive_entry *ae;
+	char buff[8488];
+	uint32_t computed_crc = 0;
+	uint32_t expected_crc = 0xf7ed24e7;
+
+	assert((a = archive_read_new()) != NULL);
+
+	extract_reference_file(refname);
+
+	assertEqualInt(ARCHIVE_OK, archive_read_free(a));
+	assert((a = archive_read_new()) != NULL);
+	assertEqualIntA(a, ARCHIVE_OK, archive_read_support_filter_all(a));
+	assertEqualIntA(a, ARCHIVE_OK, archive_read_support_format_all(a));
+
+	assertEqualIntA(a, ARCHIVE_OK,
+		archive_read_open_filename(a, refname, 10240));
+
+	assertEqualIntA(a, ARCHIVE_OK, archive_read_next_header(a, &ae));
+	assertEqualInt((AE_IFREG | 0775), archive_entry_mode(ae));
+	assertEqualString("hw-riscv64", archive_entry_pathname(ae));
+	assertEqualInt(sizeof(buff), archive_entry_size(ae));
+	assertEqualInt(sizeof(buff), archive_read_data(a, buff, sizeof(buff)));
+
+	computed_crc = crc32(computed_crc, buff, sizeof(buff));
+	assertEqualInt(computed_crc, expected_crc);
+
+	assertEqualInt(1, archive_file_count(a));
+
+	assertEqualIntA(a, ARCHIVE_EOF, archive_read_next_header(a, &ae));
+
+	assertEqualInt(ARCHIVE_OK, archive_read_close(a));
+	assertEqualInt(ARCHIVE_OK, archive_read_free(a));
+}
+#endif
+
+DEFINE_TEST(test_read_format_7zip_lzma2_riscv)
+{
+#ifdef HAVE_LZMA_FILTER_RISCV
+	struct archive *a;
+
+	assert((a = archive_read_new()) != NULL);
+
+	if (ARCHIVE_OK != archive_read_support_filter_lzma(a)) {
+			"7zip:lzma decoding is not supported on this platform");
+	} else {
+		test_riscv_filter("test_read_format_7zip_lzma2_riscv.7z");
+	}
+#else
+	skipping("This version of liblzma does not support LZMA_FILTER_RISCV");
```

Comment:
None of the CI jobs have a new-enough xz-utils to test this.

If we add the new 5.6.3 release to https://github.com/libarchive/xz I think we would be able to test this on windows, at least:
https://github.com/tukaani-project/xz/releases/tag/v5.6.3

Reply:
@mostynb I have pushed the code in https://github.com/libarchive/xz

Reply:
Thanks- let's see how upgrading that by itself goes first: https://github.com/libarchive/libarchive/pull/2417

Reply:
@mostynb thank you very much for the update!